### PR TITLE
add new flag -g to compress gifs

### PR DESCRIPTION
**Pull request by mjkkirschner**
This PR fixes an issue where animated gifs were not optimized at all. There are a few issues.
* gifs/animated gifs are quite complex - if they have been optimized before using certain methods -they cannot be optimized again - for example, if they contain frames with different internal sizes. (an optimization technique)
it's possible to undo this and then optimize with image magick, but I found this to be too slow ~ 10 minutes for 40 gifs.
Instead I found that using quantization (color reduction to 32k colors) works very well, is quite fast, and produces results about 10% - 50% reduction per gif.

* running the tool on RevitNodes.dll and Revit dictionary takes 33 seconds before this change, and 99 seconds after this change - the folder is reduced from 44megs to 38 megs. (we don't have that many .gifs in the Revit dictionary content today, only 40.)

* because reducing the .gif's is still quite slow overall, and ⚠️ ❗   **not lossless** I added a separate flag for it, this is handy for quickly debugging other parts of the tool or avoiding the gif image changes.

* also now in debug mode leave console open until console.readline for easier log reading.

this is a pathologically bad example because it has a ton of colors used in the animation.

1Meg
OverrideInView

900kB
Revit Elements Element OverrideInView_img

### CODE REVIEW

**Comment by SHKnudsen** [comment]
LGTM

**Review by SHKnudsen — APPROVED** [review]

File: src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs [hunk 1]
```diff
@@ -169,37 +170,62 @@ private static string GetDefaultContent(string nodeName)
...
+                        using (MagickImageCollection images = new MagickImageCollection(imageFileInfo))
+                        {
+                            //I've disabled these optimization steps for now because they are extremely slow.
+                            //quantization is much faster and produces mostly very good results. (though not lossless)
+
+                            //images.Coalesce();
+                            //images.Optimize();
+                            //images.OptimizeTransparency();
+                            //reduce color bit depth from 8 to 5.
+                            images.Quantize(new QuantizeSettings() { Colors = 32768 });
```
Comment (pinzart90) [inline src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs:208]: Why `32768` ?
any special meaning ?

Comment (pinzart90) [reply src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs:208]: 15-bit high color ?

**Review by pinzart90 — APPROVED** [review]
LGTM

File: src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs:208]: yes (2^5) bits per red, green, and blue `32*32*32` = 32k colors.  Normal 8 bit per channel can support: 16777216 colors.

**Comment by aparajit-pratap** [comment]
@mjkkirschner I thought that gifs never worked for documentation. Is it that after this change they now can be rendered or were they always supported and this is only for compression?

**Pull request merged**